### PR TITLE
update to have JS provied prolicy for upscale

### LINKED ISSUE #165: Need policy to control whether upscaling is enabled
For example, some decoders may have min limits, expressed in imageattr. By default, upscaling can be used for this, but there should be some control to turn this off (e.g. don't want to send "HD" video that is just upscaled to HD)

### PR DESCRIPTION
**Pull request by fluffy**
Fixes #165

### CODE REVIEW

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -621,12 +621,15 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
             attribute, it MUST first check to see if the original resolution
             meets the criteria specified in the attribute, and transmit it
             untouched if so. If the original resolution is too large for the
-            attribute criteria, the sender SHOULD apply downscaling to the
+            attribute criteria, the sender downscales the
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: This seems an unnecessary change. If you think the policy is wrong, let's debate that.

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -621,12 +621,15 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
             output of the MediaStreamTrack in order to satisfy the criteria.
             In rare cases, where a receiver requires a minimum resolution which
             is greater than the native resolution of the video, the
-            sender SHOULD apply upscaling in order to provide that resolution.
-            The sender SHOULD NOT apply upscaling in any other cases.
+            sender SHOULD apply upscaling in order to provide that resolution if
+            allowed by the upscale policy set by the Javascript and MUST not upscale if
+            the policy does not allow it.
+            The sender SHOULD NOT apply upscaling in any other cases and MUST
+            NOT if the JS policy does not allow it.
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Suggested rewrite:
"The sender MUST not apply upscaling if the upscaling policy as set by the JavaScript
forbids it. Otherwise, if the receiver requires ..." and then the original text.

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -623,10 +623,11 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
             output of the MediaStreamTrack in order to satisfy the criteria.
-            In rare cases, where a receiver requires a minimum resolution which
-            is greater than the native resolution of the video, the
-            sender SHOULD apply upscaling in order to provide that resolution.
-            The sender SHOULD NOT apply upscaling in any other cases.
+            The sender MUST not apply upscaling if the upscaling policy as set
+            by the JavaScript forbids it. Otherwise, in rare cases, where a
+            receiver requires a minimum resolution which is greater than the
+            native resolution of the video, the sender SHOULD apply upscaling in
+            order to provide that resolution.
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: You lost "The sender SHOULD NOT apply upscaling in any other cases."

**Comment by ekr** [comment]
LGTM with the change indicated above.

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -623,10 +623,13 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-            In rare cases, where a receiver requires a minimum resolution which
-            is greater than the native resolution of the video, the
-            sender SHOULD apply upscaling in order to provide that resolution.
-            The sender SHOULD NOT apply upscaling in any other cases.
+            The sender MUST not apply upscaling if the upscaling policy as set
+            by the JavaScript forbids it. Otherwise, in rare cases, where a
+            receiver requires a minimum resolution which is greater than the
+            native resolution of the video, the sender SHOULD apply upscaling in
+            order to provide that resolution. The sender SHOULD NOT apply
+            upscaling in any other cases and MUST NOT if the JS policy does not
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: This just seems duplicative. The idea is to have the logic:

If (!policy) {
  don't upscale
} else {
  if (minres > native_res) {
    SHOULD upscale
  } else {
    SHOULD NOT
  }
}

**Comment by juberti** [comment]
I rearranged the text a bit to make the policy concept more clear. PTAL.

**Comment by ekr** [comment]
LGTM

**Comment by fluffy** [comment]
LGTM
